util/docker/maketest.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+# Runs tests and reports on them to github
+
+DIR=$(cd $(dirname $0) ; pwd -P)
+source ${DIR}/docker_common.sh
+
+# Get sha sum
+USER="jgkamat"
+SUCCESS=true
+LINK_PREFIX="https://circle-artifacts.com/gh/RoboJackets/robocup-software/$CIRCLE_BUILD_NUM/artifacts/0$CIRCLE_ARTIFACTS/"
+ARTIFACT_DIR="/tmp/build_artifacts"
+
+start_pending() {
+    curl -u $USER:$TOKEN -X POST https://api.github.com/repos/robojackets/robocup-software/statuses/${SHA_SUM} -H "Content-Type: application/json" -d '{"state":"pending", "description": "A check for compiling", "context": "circle/compile", "target_url": '"\"${LINK_PREFIX}makeoutput.txt\"}"
+    curl -u $USER:$TOKEN -X POST https://api.github.com/repos/robojackets/robocup-software/statuses/${SHA_SUM} -H "Content-Type: application/json" -d '{"state":"pending", "description": "A check for cpp tests", "context": "circle/test-cpp", "target_url": '"\"${LINK_PREFIX}testcppoutput.txt\"}"
+    curl -u $USER:$TOKEN -X POST https://api.github.com/repos/robojackets/robocup-software/statuses/${SHA_SUM} -H "Content-Type: application/json" -d '{"state":"pending", "description": "A check for style", "context": "circle/style", "target_url": '"\"${LINK_PREFIX}firmwareoutput.txt\"}"
+    curl -u $USER:$TOKEN -X POST https://api.github.com/repos/robojackets/robocup-software/statuses/${SHA_SUM} -H "Content-Type: application/json" -d '{"state":"pending", "description": "A check for firmware", "context": "circle/firmware", "target_url": '"\"${LINK_PREFIX}TODONOTDONE\"}"
+}
+
+if [ "$1" = "" ]; then
+    echo "Token needed to update statuses"
+    exit 1
+elif [ "$1" = "--pending" -a "$2" = "" ]; then
+    echo "Token needed to update statuses"
+    exit 1
+elif [ "$1" = "--pending" -a "$2" != "" ]; then
+    TOKEN="$2"
+    start_pending
+    exit 0
+else
+    TOKEN="$1"
+fi
+
+# Go to root so we can make.
+cd ${ROBOCUP_ROOT}
+
+# Clean
+make clean
+git submodule update --init
+sudo chown -R `whoami`:`whoami` ${HOME}/.ccache
+
+start_pending
+
+make | tee "${ARTIFACT_DIR}/makeoutput.txt"
```

Comment:
It might be nice to put this type of stuff in a "ci task" function, then just call it at the bottom for each of the things we test.  It could take in a shortname (for logfile name and status context), a description, and a command to run.